gpii/node_modules/matchMakerFramework/src/MatchMakerUtilities.js

```diff
@@ -623,6 +623,60 @@ var fluid = fluid || require("infusion"),
         return inferred;
     };
 
+    /**
+     * Given an inferred configuration this function creates a listing of "allowed" preferences and settings,
+     * meaning a list of settings and preferences present in the inferred configuration. Any common term
+     * encountered in the inferred configuration is added to the list as top level keys with a value of true. Any application
+     * specfic term will be added under a top level key of: "http://registry.gpii.net/applications/<app-id>". The
+     * resulting "allowed" list will be built based on preferences and settings from all the "contexts" in the
+     * inferred configuration (i.e. all the contexts of the preferences set)
+     */
+    gpii.matchMakerFramework.utils.extractAllowedPreferences = function (inferredConfiguration) {
+        var allowed = {};
+        fluid.each(inferredConfiguration, function (context) {
+            fluid.each(context.applications, function (app, appId) {
+                allowed["http://registry.gpii.net/applications/" + appId] = {}; // mark the solution in allowed
+                fluid.each(app.settings, function (val, term) {
+                    if (term.startsWith("http://registry.gpii.net/applications/")) {
+                        allowed[term] = val; // copy all application specific values to a flat application block key
+                        return;
+                    }
+                    allowed[term] = true;
+                });
+            });
+        });
+        return allowed;
+    };
+
+    /**
+     * Filter a preference set based on the preferences available in the inferred configuration. This is used to
+     * ensure that no preferences are passed in the NP set that are not already given via the inferred configuration,
+     * which is useful for the something like the untrusted flowmanager model.
```

Comment:
Add big health warning that the preferences will be destructively modified by the function. Also some doc comments

Reply:
Ah well ignore that - I now see that they are not modified!